common/app-starters-task-launch-request-common/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/DataFlowTaskLaunchRequestAutoConfiguration.java

```diff
@@ -23,144 +23,139 @@
-		}
-		return builder;
+	@Bean
+	public TaskLaunchRequestSupplier taskLaunchRequestInitializer(
+			DataflowTaskLaunchRequestProperties taskLaunchRequestProperties){
+		return new DataflowTaskLaunchRequestPropertiesInitializer(taskLaunchRequestProperties);
 	}
 
-	private TaskLaunchRequestContext taskLaunchRequestContext(Message<?> message) {
-		TaskLaunchRequestContext taskLaunchRequestContext = (TaskLaunchRequestContext) message.getHeaders()
-			.get(TaskLaunchRequestContext.HEADER_NAME);
-
-		return taskLaunchRequestContext != null ? taskLaunchRequestContext : new TaskLaunchRequestContext();
+	@Bean
+	public TaskLaunchRequestMessageProcessor taskLaunchRequestMessageProcessor(
+			TaskLaunchRequestSupplier taskLaunchRequestInitializer,
+			TaskNameMessageMapper taskNameMessageMapper,
+			CommandLineArgumentsMessageMapper commandLineArgumentsMessageMapper) {
+
+		return new TaskLaunchRequestMessageProcessor(taskLaunchRequestInitializer,
+				taskNameMessageMapper,
+				commandLineArgumentsMessageMapper);
 	}
 
-	public static class DataFlowTaskLaunchRequest {
-
-		@JsonProperty("args")
-		private List<String> commandlineArguments = new ArrayList<>();
+	static class DataflowTaskLaunchRequestPropertiesInitializer extends TaskLaunchRequestSupplier {
+		DataflowTaskLaunchRequestPropertiesInitializer(
+				DataflowTaskLaunchRequestProperties taskLaunchRequestProperties){
 
-		@JsonProperty("deploymentProps")
-		private Map<String, String> deploymentProperties = new HashMap<>();
+			this.commandLineArgumentSupplier(
+					() -> new ArrayList<>(taskLaunchRequestProperties.getArgs())
+			);
 
-		@JsonProperty("name")
-		private String taskName;
+			this.deploymentPropertiesSupplier(
+					() -> KeyValueListParser.parseCommaDelimitedKeyValuePairs(
+							taskLaunchRequestProperties.getDeploymentProperties())
+			);
 
-		public void setCommandlineArguments(List<String> commandlineArguments) {
-			this.commandlineArguments = new ArrayList<>(commandlineArguments);
+			this.taskNameSupplier(()->taskLaunchRequestProperties.getTaskName());
 		}
+	}
 
-		public List<String> getCommandlineArguments() {
-			return this.commandlineArguments;
-		}
+	static class ExpressionEvaluatingCommandLineArgsMapper implements CommandLineArgumentsMessageMapper {
+		private final Map<String,Expression> argExpressionsMap;
 
-		public void setDeploymentProperties(Map<String, String> deploymentProperties) {
-			this.deploymentProperties = deploymentProperties;
-		}
+		ExpressionEvaluatingCommandLineArgsMapper(String argExpressions) {
+			argExpressionsMap = new HashMap<>();
+			if (StringUtils.hasText(argExpressions)) {
+				SpelExpressionParser expressionParser = new SpelExpressionParser();
 
-		public Map<String, String> getDeploymentProperties() {
-			return this.deploymentProperties;
-		}
+				KeyValueListParser.parseCommaDelimitedKeyValuePairs(argExpressions).forEach(
+						(k,v)-> argExpressionsMap.put(k, expressionParser.parseExpression(v)));
```

Comment:
Is it expected that there is not going to be a `BeanFactory` access in the evaluation context?
I mean you do `expression.getValue(message))` later and you don't provide an `EvaluationContext`. 
Possible one can come from Spring Integration:
```
this.evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
```